blogs/2025/07/15/copilot-coding-agent.md

```diff
@@ -0,0 +1,89 @@
+
+### Review, comment, and iterate
+
+When the agent is done, it will request your review on it's PR and assign the PR to you. It even includes a screenshot if applicable so you can quickly verify whether it did the right thing or not at all.
+
+![PR from Coding agent shown in VS Code with a screenshot](draft-with-screenshot.png)
+
+You can review the agent’s PR, leave comments, and ask for changes - all from within VS Code. The agent will picks up your feedback, updates the PR and let you know when it's finished.
+
+My colleague @pierceboggan pointed out a fascinating workflow here - with services like Azure Static Web Apps, Vercel and Netlify that offer previews for PR's, you can review the agent's work without even having to checkout it's PR and run it locally.
+
+![Live preview of the site on Netlify but we have an error](live-preview-netlify.png)
+
+Uh oh - looks like we didn't quite get the trending working. In this case, that's ok. In fact it's good. We didn't get it because our agent doesn't have access to create the required database changes so it created a migraiton script for us instead as part of the PR.
+
+But what if we wanted to give it access to make the database changes to a dev environment? Coding agent has support for MCP servers and provided you have one for your database of choice (in this case supabse), you can [configure MCP servers for the agent on the repo](https://docs.github.com/en/copilot/how-tos/agents/copilot-coding-agent/extending-copilot-coding-agent-with-mcp). It's up to you - how autonomous do you want the agent to be?
+
+![MCP configuration shown for copilot coding agent on github.com](coding-agent-mcp-config.png)
+
+But it gets even better - because you can do all of this right from the agent in VS Code.
+
+### Start sessions from Copilot Chat
+
+We find ourselves wanting to do more and more right from the chat in VS Code. This is where most work and thinking happens and you're not locked into the issue workflow. You can, at any point in a chat - no matter how long, delegate the task directly to the coding agent. All of the context that is currently a part of that chat will get sent to the coding agent.
+
+The coding agent then opens a PR and goes directly to work - skipping the issue step entirely. It distills down the context into a detailed PR description and you'll see that Copilot Coding Agent makes use of todo lists so that you can see what its doing and where it is in the process at a glance.
+
+![A GIF that shows the delegate to copilot button in chat](delegate-to-coding-agent.gif)
+
+## What’s next?
+
+The Copilot Coding Agent integration in Visual Studio Code is currently in preview, but as with everything we do, we do it all in the open and you can use it all today. If you want to light all of this up in your editor, all you need to do is make sure you have the GitHub Pull Requests extension installed and add the following settings...
```

Comment:
I think we need to list 5 things that are coming to make this paragraph interesting

@rebornix @bamurtaugh

Reply:
@olguzzar

Reply:
made some tweaks to this section, would love feedback!